### ISSUE TITLE
posibility to assign random local port of ssh tunnel to database connection port

### ISSUE THREAD

**Issue by yennor** [dbeaver/dbeaver#9654]
**Is your feature request related to a problem? Please describe.**
When settings up an ssh tunne for a database connectl I can choose as local port 0 (which is the default setting). In the tooltip it tells me, that when the local port is set to 0 a random free port > 10 000 is choosen. Which is cool.
But then, how can I assign that random port to the database connection in the tab main? If I leave the port field empty it refuses to do a conneciton.

**Describe the solution you'd like**
a clear possibility to assign that random port to the database connection.

**Comment by kseniiaguzeeva** [comment]
Please try to write **localhost** in **Host** field of **Main** tab

![dbeaver_dAvZgvdYLb](https://user-images.githubusercontent.com/51405061/91519301-ddddf380-e8fa-11ea-8d86-4590cdaf91d0.png)

**Comment by yennor** [comment]
that's what I did obviously. otherwise nothing would work. the problem is the port, not the host.
As you can see here. if i set the local port to 0 it's supposed to choose a random free port. Which is a great feature. No more error message because of already used ports.
![image](https://user-images.githubusercontent.com/6601243/91572675-526d5e00-e90c-11ea-859e-93e7f72afdfc.png)

But I don't manage to tell him in the main tab, to use that random port.

**Comment by yennor** [comment]
ha! I've just tried it out again. setting the port to 0 in the main tab actually works now.
Last time I've tried that out, it greyed out the "Test connection" button and refused to accept that port.
So it seems like its working principally, but under certain conditions doesn't accept the 0 in the main tab? Anyway it would be cool if it would be written explicitely in the pop up, that also the port in the main tab needs to be set to zero for the feature to work.
